src/System.Management/src/System/Management/InteropClasses/WMIInterop.cs

```diff
@@ -0,0 +1,2351 @@
+            int res =  WmiNetUtilsHelper.QualifierPut_f(4, pWbemQualifierSet, wszName, ref pVal, lFlavor);
+            GC.KeepAlive ( this ) ;
+            return res ;
+        }
+        public int Delete_(string wszName)
+        {
+            if(pWbemQualifierSet == IntPtr.Zero)
+                throw new ObjectDisposedException(name);
+            int res = WmiNetUtilsHelper.QualifierDelete_f(5, pWbemQualifierSet, wszName);
+            GC.KeepAlive ( this ) ;
+            return res ;
+        }
+        public int GetNames_(Int32 lFlags, out string[] pNames)
+        {
+            if(pWbemQualifierSet == IntPtr.Zero)
+                throw new ObjectDisposedException(name);
+            int res = WmiNetUtilsHelper.QualifierGetNames_f(6, pWbemQualifierSet, lFlags, out pNames);
+            GC.KeepAlive ( this ) ;
+            return res ;
+        }
+        public int BeginEnumeration_(Int32 lFlags)
+        {
+            if(pWbemQualifierSet == IntPtr.Zero)
+                throw new ObjectDisposedException(name);
+            int res = WmiNetUtilsHelper.QualifierBeginEnumeration_f(7, pWbemQualifierSet, lFlags);
+            GC.KeepAlive ( this ) ;
+            return res ;
+        }
+        public int Next_(Int32 lFlags, out string pstrName, out object pVal, out Int32 plFlavor)
+        {
+            if(pWbemQualifierSet == IntPtr.Zero)
+                throw new ObjectDisposedException(name);
+            int res = WmiNetUtilsHelper.QualifierNext_f(8, pWbemQualifierSet, lFlags, out pstrName, out pVal, out plFlavor);
+            GC.KeepAlive ( this ) ;
+            return res ;
+        }
+        public int EndEnumeration_()
+        {
+            if(pWbemQualifierSet == IntPtr.Zero)
+                throw new ObjectDisposedException(name);
+            int res = WmiNetUtilsHelper.QualifierEndEnumeration_f(9, pWbemQualifierSet);
+            GC.KeepAlive ( this ) ;
+            return res ;
+        }
+    }
+
+    class MarshalWbemObject : ICustomMarshaler 
+    {
+        public static ICustomMarshaler GetInstance(string cookie)
+        {
+            return new MarshalWbemObject(cookie);
+        }
+
+        string cookie;
+        MarshalWbemObject(string cookie)
+        {
+            this.cookie = cookie;
+        }
+
+        public void CleanUpManagedData(object obj) 
+        {
+        }
+
+        public void CleanUpNativeData(IntPtr pObj) 
+        {
+        }
+
+        public int GetNativeDataSize() 
+        {
+            return -1; // not a value type, so use -1
+        }
+
+        public IntPtr MarshalManagedToNative(object obj) 
+        {
+            return (IntPtr)obj;
+        }
+
+        public object MarshalNativeToManaged(IntPtr pObj) 
+        {
+            return new IWbemClassObjectFreeThreaded(pObj);
+        }
+    }
+    #endregion
+
+    #region Interfaces
+    [InterfaceTypeAttribute(0x0001)]
+    ///[TypeLibTypeAttribute(0x0200)]
```

Comment:
Delete?

Reply:
This one I commented, the legacy code was with it. I didn't investigate why it didn't find this attribute but it was ok with the others, I am planning to look at it.